src/blight/actions/bitcode_extract.py

```diff
@@ -0,0 +1,40 @@
+"""
+The `BitcodeExtract` action.
+"""
+
+import logging
+import os
+import subprocess
+from typing import List
+
+from blight.action import CompilerAction
+from blight.enums import Lang
+from blight.tool import CompilerTool
+
+logger = logging.getLogger(__name__)
+
+
+class BitcodeExtract(CompilerAction):
+    """
+    Compile and extract bitcode. The output bitcode file will be located in the same direcory
+    as the output executable.
+    """
+
+    def before_run(self, tool: CompilerTool) -> None:  # type: ignore
+        if tool.lang in [Lang.C, Lang.Cxx]:
+            args: List[str]
+            if tool.outputs:
+                args = ["-c", "-emit-llvm", "-o", tool.outputs[0] + ".bc", *tool.inputs]
```

Comment:
I *think* this needs to be redesigned slightly. Let's say you have an invocation like this:

```bash
cc -o some_exe foo.c bar.c baz.c
```

With your current implementation, this would generate the following bitcode:

```bash
cc -c -emit-llvm -o some_exe.bc foo.c bar.c baz.c -flto
```

whereas what I _think_ we want is something like this:

```bash
cc -c -S -emit-llvm -o foo.bc foo.c
cc -c -S -emit-llvm -o bar.bc bar.c
# etc
```

This latter approach avoids the need for LTO, which might give LLVM optimization opportunities that we don't necessarily want to enable (in the interest of eventually recompiling and producing binaries that look as close to the original as possible).

Reply:
On a similar note: I think it makes sense to write our bitcode outputs to a custom directory, to avoid polluting the build itself. The `FindOutputs` action has an example of using a `store` setting in its config to create an intermediate directory; that might be a good pattern for this action as well.

(We also need to figure out how best to associate each individual bitcode output with its source input -- WLLVM and GLLVM do this by injecting information with `objcopy`. But we can punt on that for this initial PR.)